virttest/libvirt_xml/vm_xml.py

```diff
@@ -538,21 +538,30 @@ def del_controller(self, controller_type=None):
 
         :return: None if deleting all controllers
         """
-        all_controllers = self.xmltreefile.findall("devices/controller")
-        del_controllers = []
-        for controller in all_controllers:
-            if controller.get("type") != controller_type:
-                continue
-            del_controllers.append(controller)
-
         # no seclabel tag found in xml.
+        del_controllers = self.get_controllers(self, controller_type=None)
```

Comment:
`del_controllers = self.get_controllers() `, I guess we do not need a `self` and `controller_type` is anyways the default value.

Reply:
Fixed, thanks